services/httppost/service.go

```diff
@@ -310,5 +316,21 @@ func (h *handler) Handle(event alert.Event) {
 		h.diag.Error("failed to POST alert data", err)
 		return
 	}
-	resp.Body.Close()
+	defer resp.Body.Close()
+
+	if resp.StatusCode/100 != 2 {
+		var err error
+		if h.captureResponse {
+			var body []byte
+			body, err = ioutil.ReadAll(resp.Body)
+			if err == nil {
+				// Use the body content as the error
+				err = errors.New(string(body))
+			}
+		} else {
+			err = errors.New("unknown error, use .captureResponse() to capture the HTTP response")
+		}
+		h.diag.WithContext(keyvalue.KV("code", strconv.Itoa(resp.StatusCode))).Error("POST returned non 2xx status code", err)
```

Comment:
`WithContext` returns a new diagnostic. Might be worth it to change the `Error` method on `Diagnostic` to

```go
Error(msg string, err error, ctx ...keyvalue.T)
```

Reply:
will do

Reply:
Fixed